src/microservices/Microservices.IsIdentifiable/Service/IsIdentifiableQueueConsumer.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using Microservices.IsIdentifiable.Reporting;
+using NLog;
+using RabbitMQ.Client.Events;
+using Smi.Common.Messages;
+using Smi.Common.Messages.Extraction;
+using Smi.Common.Messaging;
+using Newtonsoft.Json;
+
+namespace Microservices.IsIdentifiable.Service
+{
+    public class IsIdentifiableQueueConsumer : Consumer, IDisposable
+    {
+        private readonly IProducerModel _producer;
+        private readonly string _fileSystemRoot;
+        private readonly IClassifier _classifier;
+
+        public IsIdentifiableQueueConsumer(IProducerModel producer, string fileSystemRoot, IClassifier classifier)
+        {
+            _producer = producer;
+            _fileSystemRoot = fileSystemRoot;
+            _classifier = classifier;
+        }
+
+        protected override void ProcessMessageImpl(IMessageHeader header, BasicDeliverEventArgs basicDeliverEventArgs)
+        {
+            // Deserialize the message from the delivery arguments
+            if (!SafeDeserializeToMessage(header, basicDeliverEventArgs, out ExtractFileStatusMessage message))
+                return;
+
+            // The path is taken from the message, however maybe it should be FileSystemOptions|ExtractRoot in default.yaml
+            // If the filename has a rooted path then the ExtractionDirectory is ignored by Path.Combine
+            var toProcess = new FileInfo( Path.Combine(message.ExtractionDirectory, message.AnonymisedFileName) );
+
+            if(!toProcess.Exists)
```

Comment:
Do we really want to completely crash out here?

Reply:
What would you like? Nack? or output message support for missing files? either would be fine by me

Reply:
Just checking, but I think the CTP service handles a missing file by sending a "failure" message. We could do the same here?

This is hopefully not a common occurrence, but I'd rather not kill the service or leave the CohortPackager wondering why it's missing an image from its final accounting.

Reply:
CTP currently does this: https://github.com/SMI/SmiServices/blob/d0c21df7f04c2d2365b6ab88ab09033352ced303/src/microservices/com.smi.microservices.ctpanonymiser/src/main/java/org/smi/ctpanonymiser/messaging/CTPAnonymiserConsumer.java#L73

Reply:
Resolved offline